packages/core/src/fhirpath/utils.ts

```diff
@@ -296,8 +299,25 @@ export function fhirPathEquivalent(x: TypedValue, y: TypedValue): TypedValue[] {
   if (isQuantity(xValue) && isQuantity(yValue)) {
     return booleanToTypedValue(isQuantityEquivalent(xValue, yValue));
   }
+
+  if (xType === 'Coding' && yType === 'Coding') {
+    if (typeof xValue !== 'object' || typeof yValue !== 'object') {
+      return booleanToTypedValue(false);
+    }
+    // "In addition, for Coding values, equivalence is defined based on the code and system elements only.
+    // The version, display, and userSelected elements are ignored for the purposes of determining Coding equivalence."
+    // Source: https://hl7.org/fhir/fhirpath.html#changes
+
+    // We need to check if both `code` and `system` are equivalent.
+    // If both have undefined `system` fields, If so, then the two's `system` values must be compared.
+    // Essentially they must both be `undefined` or both the same.
+    return booleanToTypedValue(
+      (xValue as Coding).code === (yValue as Coding).code && (xValue as Coding).system === (yValue as Coding).system
+    );
+  }
+
   if (typeof xValue === 'object' && typeof yValue === 'object') {
-    return booleanToTypedValue(deepEquals(xValue, yValue));
+    return booleanToTypedValue(deepEquals({ ...xValue, id: undefined }, { ...yValue, id: undefined }));
```

Comment:
@ThatOneBro  Is this in the spec somewhere?

Reply:
Woah!  TIL: https://build.fhir.org/fhirpath.html#changes

"Equivalence works in exactly the same manner, but with the addition that for complex types, equality requires all child properties to be equal, except for "id" elements.'

Reply:
Yes it is, maybe I should add a comment about that too...

Here's what the spec says:
`Equivalence works in exactly the same manner, but with the addition that for complex types, equality requires all child properties to be equal, except for "id" elements.`

https://hl7.org/fhir/fhirpath.html#changes